populate-gems-cache/action.yml

```diff
@@ -0,0 +1,34 @@
+name: Populate Bundler Gems Cache
+description: Populates the Homebrew Bundler RubyGems Cache
+runs:
+  using: "composite"
+  steps:
+    - name: Set up Homebrew
```

Comment:
Feels a bit weird that this will end up meaning you remove `Homebrew/actions/setup-homebrew` from those workflows. Instead, how about making this an option on `Homebrew/actions/setup-homebrew` instead? Could default to off by default and then default to on by default later; feels like most people will want this most of the time.